CMSgov/design-system
# [WNMGDS 834] button with analytics

## Summary
Adding optional Google Analytics to button component

### Added
Google Analytics data to button component that is sent when button is clicked

### Fixed
Updated typescript definitions for prop types

## How to test
http://design-system-demo.s3-website-us-east-1.amazonaws.com/WNMGDS-834/button-with-analytics

1. On line 17 of file `design-system/packages/design-system/src/components/flags.ts`, change value to `true` to turn on analytics for button component
2. Run the site locally and navigate to the Button component and the React example section
3. Ensure that you have the Omnibug Chrome extension installed. Click a button and ensure that the event is captured by the Omnibug extension
4. Ensure that values for `Event Action`, `Event Category` and `Event Label` are all what you'd expect
5. Check out the Button component documentation and ensure that all analytics information is available and understandable.
